File: frontend/src/api/autoTagging.ts
import { api } from './client'
import type { AutoTaggingStatus } from './types'

export function fetchAutoTaggingStatus(): Promise<AutoTaggingStatus> {
  return api.get<AutoTaggingStatus>('/api/v1/auto-tagging')
}

export function updateAutoTagging(enabled: boolean): Promise<AutoTaggingStatus> {
  return api.patch<AutoTaggingStatus>('/api/v1/auto-tagging', { enabled })
}

export function retrainAutoTagging(): Promise<AutoTaggingStatus> {
  return api.post<AutoTaggingStatus>('/api/v1/auto-tagging/retrain')
}

File: backend/src/api/v1/auto_tagging.py
from fastapi import APIRouter
from src.core.dependencies import AutoTaggingSvc, CurrentUser
from src.schemas.auto_tagging import (
    AutoTaggingSettingsUpdate,
    AutoTaggingStatusResponse,
)

router = APIRouter(prefix="/api/v1/auto-tagging", tags=["auto-tagging"])


@router.get("", response_model=AutoTaggingStatusResponse)
async def get_auto_tagging_status(
    current_user: CurrentUser,
    service: AutoTaggingSvc,
) -> AutoTaggingStatusResponse:
    return await service.status(current_user)


@router.patch("", response_model=AutoTaggingStatusResponse)
async def update_auto_tagging(
    data: AutoTaggingSettingsUpdate,
    current_user: CurrentUser,
    service: AutoTaggingSvc,
) -> AutoTaggingStatusResponse:
    return await service.set_enabled(current_user, data.enabled)


@router.post("/retrain", response_model=AutoTaggingStatusResponse)
async def retrain_auto_tagging(
    current_user: CurrentUser,
    service: AutoTaggingSvc,
) -> AutoTaggingStatusResponse:
    return await service.retrain(current_user)


@router.get("/metrics", response_model=AutoTaggingStatusResponse)
async def get_auto_tagging_metrics(
    current_user: CurrentUser,
    service: AutoTaggingSvc,
) -> AutoTaggingStatusResponse:
    return await service.status(current_user)
